vispy/scene/cameras/cameras.py

```diff
@@ -71,43 +197,191 @@ def interactive(self):
         return self._interactive
     
     @interactive.setter
-    def interactive(self, b):
-        self._interactive = b
-
+    def interactive(self, value):
+        self._interactive = bool(value)
+    
     @property
-    def viewbox(self):
-        """ The ViewBox that this Camera is attached to.        
-        """
-        return self._viewbox
-    
-    @viewbox.setter
-    def viewbox(self, vb):
-        if self._viewbox is not None:
-            self.disconnect()
-        self._viewbox = vb
-        if self._viewbox is not None:
-            self.connect()
-            self.parent = vb.scene
+    def scale_factor(self):
+        """ The measure for the scale or range that the camera should cover
+        
+        For the PanZoomCamera and TurnTableCamera this translates to
+        zooming: set to smaller values to zoom in.
+        """ 
+        return self._scale_factor
+    
+    @scale_factor.setter
+    def scale_factor(self, value):
+        self._scale_factor = abs(float(value))
         self.view_changed()
+    
+    @property
+    def aspect_ratio(self):
+        """ The aspect ratio to show the scene with
         
-    def view_changed(self):
-        """ Called when this camera is connected to a new view.
+        E.g. (0.5, 1.0, 1.0) would make the x-dimension half as small.
+        Negative values can be used to flip dimensions, e.g. (1, -1)
+        is typically suited for visualizing images.
+        """
+        return self._aspect_ratio
+    
+    @aspect_ratio.setter
+    def aspect_ratio(self, value):
```

Comment:
`scale_ratio` seems appropriate. It also seems more consistent with `scale_factor`.